### PR TITLE
[New Docs] Remove entries under Old Docs for content that has been updated.

### PR DESCRIPTION
**Pull request by hramos**
All the content in these old guides has been moved elsewhere.

Also updated the ids/permalinks for the Reference guides to be consistent.

### CODE REVIEW

File: docs/docs-old/08-working-with-the-browser.md [hunk 1]
```diff
@@ -19,42 +19,3 @@ Most of the time you should stay within React's "faked browser" world since it's
...
-
-* `componentWillUnmount()` is invoked immediately before a component is unmounted and destroyed. Cleanup should go here.
-
-### Mounted Methods
-
-_Mounted_ composite components also support the following method:
-
-* `component.forceUpdate()` can be invoked on any mounted component when you know that some deeper aspect of the component's state has changed without using `this.setState()`.
-
-## Browser Support
```
Comment (gaearon) [inline docs/docs-old/08-working-with-the-browser.md]: Can we add this to React DOM documentation page?

Comment (hramos) [inline docs/docs-old/08-working-with-the-browser.md]: Good catch

File: docs/docs/reference-react-dom.md [hunk 2]
```diff
@@ -9,6 +9,14 @@ permalink: docs/react-dom.html
 
 The `react-dom` package provides DOM-specific methods that can be used at the top level of your app and as an escape hatch to get outside of the React model if you need to. Most of your components should not need to use this module.
 
+### Browser Support
+
+React supports most popular browsers, including Internet Explorer 9 and above.
```
Comment (gaearon) [inline docs/docs/reference-react-dom.md]: Can we say "all popular browsers"? I don't think IE8 is popular at this point.

File: docs/docs/reference-react-dom.md [hunk 3]
```diff
@@ -9,6 +9,14 @@ permalink: docs/react-dom.html
...
+
+> Note
+>
+> We don't support older browsers that don't support ES5 methods, but you may find that your apps do work in older browsers if polyfills such as [es5-shim and es5-sham](https://github.com/es-shims/es5-shim) are included in the page. You're on your own if you choose to take this path.
+
  - [`render()`](#render)
```
Comment (gaearon) [inline docs/docs/reference-react-dom.md]: Maybe make this section last? Now it looks like APIs like `render()` are grouped under it.

**Comment by hramos** [comment]
Feedback taken into account, PTAL.

File: docs/docs/reference-react-dom.md (hunk 3 above)
Comment (hramos) [inline docs/docs/reference-react-dom.md]: Done.

File: docs/docs/reference-react-dom.md (hunk 2 above)
Comment (hramos) [inline docs/docs/reference-react-dom.md]: Done.

File: docs/docs-old/08-working-with-the-browser.md (hunk 1 above)
Comment (hramos) [inline docs/docs-old/08-working-with-the-browser.md]: Done.

**Pull request merged**